src/main/java/com/redhat/pantheon/asciidoctor/AsciidoctorPool.java

```diff
@@ -44,9 +44,14 @@ public AsciidoctorPool(@Reference GlobalConfig globalConfig) {
      */
     public Asciidoctor borrowObject(Resource base) {
         Asciidoctor asciidoctor = super.borrowObject();
-        asciidoctor.javaExtensionRegistry().includeProcessor(
-                new SlingResourceIncludeProcessor(base));
-        return asciidoctor;
+        try {
+            asciidoctor.javaExtensionRegistry().includeProcessor(
+                    new SlingResourceIncludeProcessor(base));
+            return asciidoctor;
+        } catch (Exception e) {
+            returnObject(asciidoctor);
```

Comment:
Hhmm... this is an interesting scenario. It actually reaffirms my position in my previous comment. The only reason for this overloaded method is so we can introduce the proper asciidoctor extensions. If we leave that responsibility to the caller, this method becomes unnecessary.